### ISSUE TITLE
GetTypeInfo call is ambiguous

### ISSUE THREAD

**Issue by slozier** [JoshClose/CsvHelper#514]
It looks like your corefx compatibility extension methods are conflicting with .NET Framework 4.5 methods. Perhaps you could make them internal?

I get the following error message:
```
The call is ambiguous between the following methods or properties: 'System.Reflection.IntrospectionExtensions.GetTypeInfo(System.Type)' and 'System.Reflection.TypeExtensions.GetTypeInfo(System.Type)'
```

To replicate, create a project targetting .NET Framework 4.5, add CsvHelper as a NuGet package:
```
using System.Reflection;

namespace ConsoleApplication
{
    class Program
    {
        static void Main(string[] args)
        {
            var tmp = typeof(string).GetTypeInfo();
        }
    }
}

```

**Comment by michaelahern** [comment]
+1

**Comment by JoshClose** [comment]
I didn't think that was a part of 4.5. I thought it was only PCL/CoreFX. I'll change it to only be a part of 4.0 and below then.

**Comment by JoshClose** [comment]
I added a net45 build and excluded the CoreFXCompatibility stuff from it.

**Comment by ggirard07** [comment]
Still facing this issue with current version 2.16.1 when used in a PCL with Profile111 (.Net framework 4.5 compatible which already contains the System.Reflection extensions).
Reproduction is available here https://github.com/ggirard07/CsvHelperGetTypeInfoIssue

**Comment by JoshClose** [comment]
If you can find any documentation that shows what APIs are included in which version of the PCL, let me know.

**Comment by JoshClose** [comment]
I found [this](https://visualstudiogallery.msdn.microsoft.com/1177943e-cfb7-4822-a8a6-e56c7905292b) which seems to be exactly what I'm looking for. It seems to be a little flaky though.

Also [this](https://github.com/dotnet/corefx/blob/master/Documentation/architecture/net-platform-standard.md) for future reference.

**Comment by slozier** [comment]
Also http://dotnetstatus.azurewebsites.net/fxapi?docId=M%3ASystem.Reflection.IntrospectionExtensions.GetTypeInfo(System.Type)

Have you considered making your extension method internal instead?

**Comment by JoshClose** [comment]
Thanks. Making it internal would probably fix it. I'm looking into restructuring the PCL profile now. Only include things that aren't in .NET <= 4.5.1 and netstandard1.3. I think the PCL only needs to cover Silverlight 5, Windows 8, and WP 8.

**Comment by JoshClose** [comment]
This is fixed in NuGet version 2.16.2. I just made it internal.

I'm going to make the other changes to the 3.0 beta work.
